fix: Theme Card styles

vgeorge opened the pull request:
Contributes to https://github.com/NASA-IMPACT/veda-ui/issues/1666.

This should address the remaining style issues on the Theme cards. I haven't included the carrousel to the dashboard in this PR to keep the scope manageable.

Changes added:

- Introduced SmallCard to hold common styles for Exhibit and Theme Cards
- Fixed height/widht and alignment
- Updated background images
- Applied hover effect to Theme card

Preview:

![theme](https://github.com/user-attachments/assets/a355a3b5-654e-4fc9-a834-17796a28e4d9)

To give feedback, I suggest reviewing the listed items in the upstream ticket.

dzole0311 commented on app/components/common/cards/ThemeCard.tsx:
```diff
@@ -1,39 +1,61 @@
-        <CardBadge label='Theme' icon='Language' />
-        <h2 className='position-absolute bottom-0 width-full padding-2 bg-white-90 margin-0 text-white'>
-          {theme.title}
+    <SmallCard key={theme.id} className='theme-card'>
```
Would it make more sense to keep `ThemeCard` (and all other cards) fully responsive at 100% width and handle layout via a parent Grid in the consumer pages?

Basically removing the `SmallCard` and the `gridLayout` prop from the `ThemeCard` and just using the grid system to arrange the cards in a way we want throughout the pages.

I noticed that hardcoding the max-width causes layout gaps between breakpoints (tablets, mobile), like the extra space on the right here:

<img width="819" alt="Screenshot 2025-05-19 at 15 33 27" src="https://github.com/user-attachments/assets/12cea908-ad42-4041-8c71-cae65135786b" />

vs this (w/o the max-width of 22rem)

<img width="814" alt="Screenshot 2025-05-19 at 15 33 36" src="https://github.com/user-attachments/assets/7330eb69-921a-4d86-86fe-6f033ebdec58" />

vgeorge replied:
@dzole0311 I agree we should avoid a hardcoded value, I removed it. I'm not sure using a Grid wrapper, the `gridLayout` property now seems to be working well on all screen sizes and the output seems semantically correct when using CardGroup.

dzole0311 commented on app/components/common/cards/ThemeCard.tsx:
```diff
@@ -1,39 +1,61 @@
-          {theme.title}
+    <SmallCard key={theme.id} className='theme-card'>
+      <CustomCardMedia className='position-relative'>
+        <CardBadge label='Earth Theme' icon='Language' />
```
Can we add `font-body-3xs` in the badge component to match the designs font size? Was marked as resolved in the [previous PR](https://github.com/NASA-IMPACT/next-earth-gov/pull/33/files#r2092433369), but I can't seem to see it on dev

vgeorge replied:
I forgot to push changes to the other PR 🤦

dzole0311 commented on app/components/common/cards/ThemeCard.scss:
```diff
@@ -8,5 +8,102 @@
+  
+  &:hover {
+    .usa-card__container {
+      box-shadow: 0 8px 24px rgba(0, 0, 0, 0.15);
```
[Do we need the](https://www.figma.com/design/2Vd0SDhFOTBJaYioJPr0Ei/Earth.gov?node-id=2433-27043&t=g2yk72OiN0ZqlEMn-4) `box-shadow`?

vgeorge replied:
Removed 👍

dzole0311 reviewed (changes requested):
Looks great! 👍

vgeorge commented:
@dzole0311 thanks for the review. I addressed the comments and this is ready for another review. 

I made a refactor on the Exhibit Card overlay, but it is not complete yet and we should match the designs. Let's address in separate.

dzole0311 commented:
@vgeorge Can we keep the existing images for the themes page banners (to at least have some banner image for stakeholders reviewing the staging environment)? They seem to not [show up now](https://deploy-preview-35--next-earth-gov.netlify.app/themes/agriculture).

vgeorge commented:
@dzole0311 I reinstated the assets now. I thought they were only used by the cards.

dzole0311 reviewed (approved):
Thanks @vgeorge !
